src/scala/com/github/johnynek/bazel_deps/GradleResolver.scala

```diff
@@ -42,203 +37,214 @@ class GradleResolver(
       }
     }
 
-  private def mergeLockFiles(
-      lockFiles: List[GradleLockFile]
-  ): Try[GradleLockFile] =
-    lockFiles.foldM(GradleLockFile.empty) {
-      TryMerge.tryMerge(None, _, _)
-    }
-
   // Gradle has compile/runtime/test sepearate classpaths
   // we just want one, so we merge them all
   private def collapseDepTypes(
       lockFile: GradleLockFile
   ): Try[Map[String, GradleLockDependency]] =
-    List(
-      lockFile.compileClasspath,
-      lockFile.annotationProcessor,
-      lockFile.resolutionRules,
-      lockFile.runtimeClasspath,
-      lockFile.testCompileClasspath,
-      lockFile.testRuntimeClasspath
+    TryMerge.tryMergeAll(
+      None,
+      NonEmptyList.of(
+        lockFile.compileClasspath,
+        lockFile.annotationProcessor,
+        lockFile.resolutionRules,
+        lockFile.runtimeClasspath,
+        lockFile.testCompileClasspath,
+        lockFile.testRuntimeClasspath
+      )
     )
     .map(_.getOrElse(Map.empty))
-    .foldM(Map.empty[String, GradleLockDependency]) {
-      TryMerge.tryMerge(None, _, _)
-    }
 
   private def assertConnectedDependencyMap(
       depMap: Map[String, GradleLockDependency]
   ): Try[Unit] =  {
     // Use validated to collect all the missing errors, not just the first
-    type V[+A] = ValidatedNel[(String, String), A]
+    type V[+A] = ValidatedNec[(String, String), A]
 
-    def assertDep(key: String, gld: GradleLockDependency): ValidatedNel[(String, String), Unit] =
+    val unit = Validated.valid(())
```

Comment:
Similarly, we can lift this out farther right?

Reply:
btw: in bosatsu, constants like this are automatically lifted out of scope they can always be statically allocated.